### DISCUSSION TITLE
How to attach header in URL for AsyncImage (Swift5+)

### DISCUSSION THREAD

**Discussion by Eumag** [onevcat/Kingfisher#2014] [Q&A]
Hello, 

AsyncImage in Swift takes in URL as its parameter. With authentication, (jwt) token is required to be passed as header. For example: 
```
set TOKEN=<cognito_access_token>
curl -H "Authorization: Bearer ${TOKEN}" https://d21ivree8p8cbj.cloudfront.net/animals/IMG_2594.png
```

Usage of AsyncImage in Swift is: 
```
let urlString = "https://images.unsplash.com/photo-1581274998394-10ac73d6a122?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=987&q=80"

AsyncImage(url: URL(string: urlString)) { image in
    image.resizable()
} placeholder: {
    ProgressView()
}
.frame(width: 400, height: 400, alignment: .center)
```

How can one attach headers into this URL so that AsyncImage can download the image from source without hitting Authentication error. If need to implement own solution for such AsyncImage call, what would be structure of that - which Swift constructs can be used - any sample reference? Could delegate be helpful in a way to attach headers automatically while making AsyncImage call.

Thanks.

**Comment by onevcat** [comment]
AsyncImage is a type of SwiftUI and it is not related to this framework, Kingfisher. If you'd like to use Kingfisher to load an image from a certain URL and with the modified header, try [the `requestModifier`](https://swiftpackageindex.com/onevcat/kingfisher/master/documentation/kingfisher/kfoptionsetter/requestmodifier(_:)-13206). Also see the wiki for some example: https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#modify-a-request-before-sending (Although it is for UIKit, but the same applies to `KFImage` in SwiftUI.

**Reply by Eumag** [reply]
Thanks for suggesting on it. Checked the codebase, we can close this ticket.
